Compilation errors and app failed to start when using Micronaut + R2DBC

### Task List

- [X] Steps to reproduce provided
- [X] Stacktrace (if present) provided
- [X] Example that reproduces the problem uploaded to Github
- [ ] Full description of the issue provided (see below)

### Steps to Reproduce

1. Create a new project.
2. Annotate repository interface.
3. Perform a Gradle run: `./gradlew run`

### Expected Behaviour

The application should be started without problems.

### Actual Behaviour

Application startup failed due to a strange error : 
```
Task :compileGroovy FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileGroovy'.
> BUG! exception in phase 'semantic analysis' in source unit '/home/jk/code/workspace/bidbattle/api/bid/src/main/groovy/com/example/data/repo/UserRepository.groovy' null
```

### Environment Information

- **Operating System**: `Ubuntu 20.04.1 LTS`
- **Micronaut Version:** `micronautVersion=2.3.2-SNAPSHOT`
- **JDK Version:** 
  ```
   openjdk version "11.0.9.1" 2020-11-04
   OpenJDK Runtime Environment (build 11.0.9.1+1-Ubuntu-0ubuntu1.20.04)
   OpenJDK 64-Bit Server VM (build 11.0.9.1+1-Ubuntu-0ubuntu1.20.04, mixed mode, sharing)
   ```

### Example Application

[Example App on Github](https://github.com/jamil-kafi/Bid)

--- a/inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/GroovyClassElement.java
+++ b/inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/GroovyClassElement.java
@@ -425,7 +425,7 @@ public class GroovyClassElement extends AbstractGroovyElement implements Arrayab
                         } else {
                             type = gt.getType();
                         }
-                        AnnotationMetadata annotationMetadata = resolveAnnotationMetadata(cn);
+                        AnnotationMetadata annotationMetadata = resolveAnnotationMetadata(type);
                         typeArgumentMap.put(redirectType.getName(), new GroovyClassElement(
                                 visitorContext,
                                 type,

